wallet-new/src/Cardano/Wallet/Kernel/BListener.hs

```diff
@@ -81,11 +81,13 @@ switchToFork :: PassiveWallet
              -> IO (Either RollbackDuringRestoration ())
 switchToFork pw@PassiveWallet{..} n bs = do
     k <- Node.getSecurityParameter _walletNode
-    blockssByAccount <- mapM (prefilterBlock' pw) bs
+    blocksAndMeta <- mapM (prefilterBlock' pw) bs
+    let (blockssByAccount, metas) = unzip blocksAndMeta
     res <- update' _wallets $ SwitchToFork k n blockssByAccount
     case res of
       Left  err     -> return $ Left err
-      Right changes -> do modifyMVar_ _walletSubmission $
+      Right changes -> do mapM_ (putTxMeta _walletMeta) $ concat metas
```

Comment:
I add the reply also here: If the Tx was added by the same account it will be ignored. If it was added by a different account it will create a new TxMeta entry. In any case, if Inputs and Outputs are different it will throw an error.